dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java

```diff
@@ -1653,6 +1657,44 @@ protected String validateAppsLoaded(List<String> appList, int intendedTimeout, i
         }
     }
 
+    private void validateTestedFeatures(RemoteFile serverLog) throws Exception {
+        final String m = "validateTestedFeatures";
+        // Load the tested feature data, if it exists
+        File testedFeaturesFile = new File("fat-feature-deps.json");
+        if (!testedFeaturesFile.exists()) {
+            Log.info(c, m, "No tested feature data for this server.  Skipping feature validation");
+            return;
+        }
+
+        // Scrape messages.log to see what features were installed
+        List<String> installedFeaturesRaw = LibertyFileManager.findStringsInFile("CWWKF0012I: .*", serverLog);
+        if (installedFeaturesRaw == null || installedFeaturesRaw.size() == 0)
+            return;
+        Set<String> installedFeatures = new HashSet<String>();
+        for (String f : installedFeaturesRaw)
+            for (String installedFeature : f.substring(0, f.lastIndexOf(']')).substring(f.lastIndexOf('[') + 1).split(","))
+                installedFeatures.add(installedFeature.trim().toLowerCase());
+        Log.info(c, m, "Installed features are: " + installedFeatures);
+
+        // Make sure that any features installed in the server are defined in the fat-feature-deps list
+        JsonArray testedFeaturesJson = Json.createReader(new FileInputStream(testedFeaturesFile)).readArray();
```

Comment:
This would seem to be capable of throwing an NPE, according to the personal build.

Reply:
Did you mean to say CNFE instead?  I don't think we can get a NPE here because earlier we have checked that `testedFeaturesFile` exists

Reply:
No, I meant NPE. In the personal build the `com.ibm.ws.jsonp_fat` bucket caught an NPE here:

```
java.lang.NullPointerException
	at componenttest.topology.impl.LibertyServer.validateTestedFeatures(LibertyServer.java:1680)
	at componenttest.topology.impl.LibertyServer.validateServerStarted(LibertyServer.java:2092)
	at componenttest.topology.impl.LibertyServer.startServerWithArgs(LibertyServer.java:1349)
	at componenttest.topology.impl.LibertyServer.startServerAndValidate(LibertyServer.java:956)```